lib/maplibre-compose-expressions/src/commonMain/kotlin/dev/sargunv/maplibrecompose/expression/dsl/ramps.kt

```diff
@@ -0,0 +1,142 @@
+package dev.sargunv.maplibrecompose.expression.dsl
+
+import dev.sargunv.maplibrecompose.expression.ast.Expression
+import dev.sargunv.maplibrecompose.expression.ast.FunctionCall
+import dev.sargunv.maplibrecompose.expression.value.ColorValue
+import dev.sargunv.maplibrecompose.expression.value.ExpressionValue
+import dev.sargunv.maplibrecompose.expression.value.FloatValue
+import dev.sargunv.maplibrecompose.expression.value.InterpolateableValue
+import dev.sargunv.maplibrecompose.expression.value.InterpolationValue
+
+/**
+ * Produces discrete, stepped results by evaluating a piecewise-constant function defined by pairs
+ * of input and output values ([stops]). Returns the output value of the stop just less than the
+ * [input], or the [fallback] if the input is less than the first stop.
+ *
+ * Example:
+ * ```
+ * step(zoom(), const(0), 10 to const(2.5), 20 to const(10.5))
+ * ```
+ *
+ * returns 0 if the zoom is less than 10, 2.5 if the zoom is between 10 and less than 20, 10.5 if
+ * the zoom is greater than or equal 20.
+ */
+public fun <T : ExpressionValue> step(
+  input: Expression<FloatValue>,
+  fallback: Expression<T>,
+  vararg stops: Pair<Number, Expression<T>>,
+): Expression<T> =
+  FunctionCall.of(
+      "step",
+      input,
+      fallback,
+      *stops
+        .sortedBy { it.first.toFloat() }
+        .foldToArgs {
+          add(const(it.first.toFloat()))
+          add(it.second)
+        },
+    )
+    .cast()
+
+private fun <T, V : InterpolateableValue<T>> interpolateImpl(
+  name: String,
+  type: Expression<InterpolationValue>,
+  input: Expression<FloatValue>,
+  vararg stops: Pair<Number, Expression<V>>,
+): Expression<V> =
+  FunctionCall.of(
+      name,
+      type,
+      input,
+      *stops
+        .sortedBy { it.first.toDouble() }
+        .foldToArgs {
+          add(const(it.first.toFloat()))
+          add(it.second)
+        },
+    )
+    .cast()
+
+/**
+ * Produces continuous, smooth results by interpolating between pairs of input and output values
+ * ([stops]), given the [input] value.
+ *
+ * Example:
+ * ```
+ * interpolate(
+ *   exponential(2), zoom(),
+ *   16 to const(1),
+ *   24 to const(256),
+ * )
+ * ```
+ *
+ * interpolates exponentially from 1 to 256 in zoom levels 16 to 24. Below zoom 16, it is 1, above
+ * zoom 24, it is 256. Applied to for example line width, this has the visual effect that the line
+ * stays the same width in meters on the map (rather than on the viewport).
+ */
+public fun <T, V : InterpolateableValue<T>> interpolate(
+  type: Expression<InterpolationValue>,
+  input: Expression<FloatValue>,
+  vararg stops: Pair<Number, Expression<V>>,
+): Expression<V> = interpolateImpl("interpolate", type, input, *stops)
```

Comment:
Now that the interpolation type continues to be not a sealed class, and probably that will stay that way now(?), the documentation comments should mention which functions produce an `Expression<InterpolationValue>`